pkg/api/validation/validation.go

```diff
@@ -2426,7 +2405,7 @@ func ValidatePreferredSchedulingTerms(terms []api.PreferredSchedulingTerm, fldPa
 // validatePodAffinityTerm tests that the specified podAffinityTerm fields have valid data
 func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopologyKey bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if !utilfeature.DefaultFeatureGate.Enabled(features.AffinityInAnnotations) && len(podAffinityTerm.TopologyKey) == 0 {
+	if len(podAffinityTerm.TopologyKey) == 0 {
```

Comment:
This should be removed as it is being checked below if !allowEmptyTopologyKey && len(podAffinityTerm.TopologyKey) == 0 as validatePodAffinityTerm is always passed with allowEmptyTopologyKey as false. Or another way would be to refactor  validatePodAffinityTerm and validatePodAffinityTerms.